cnf-tests/testsuites/e2esuite/sctp/sctp.go

```diff
@@ -356,10 +359,10 @@ func testClientServerConnection(cs *client.ClientSet, namespace string, destIP s
 
 	if !shouldSucceed {
 		Consistently(func() k8sv1.PodPhase {
-			pod, err := cs.Pods(namespace).Get(context.Background(), serverPodName, metav1.GetOptions{})
+			pod, err = cs.Pods(namespace).Get(context.Background(), serverPodName, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 			return pod.Status.Phase
-		}, 30*time.Second, 1*time.Second).Should(Equal(k8sv1.PodRunning))
+		}, 30*time.Second, 1*time.Second).Should(Equal(k8sv1.PodRunning), fmt.Sprintf("pod %s failed to be in running state. Events:\n %s", pod.Name, pods.GetStringEventsForPod(cs, pod)))
```

Comment:
`pods.GetStringEventsForPod(cs, pod)` is invoked too early, before the first `Consistently` loop. You are probably looking for a lazy evaluation:

```
(...).Should(
    Equal(k8sv1.PodRunning), 
    func () string {
        return fmt.Sprintf("pod %s failed to be in running state. Events:\n %s", pod.Name, pods.GetStringEventsForPod(cs, pod)))
    })
```

And Gomega supports that:
https://onsi.github.io/gomega/#annotating-assertions

Reply:
Thanks for your comment. it really is a crucial point. fixed

Reply:
Thanks for your comment. it really is a crucial point. fixed